### DISCUSSION TITLE
How to scope provider?

### DISCUSSION THREAD

**Discussion by vimhead** [rrousselGit/riverpod#1562] [Q&A] [answered]
For example, I have a stack of UserPage (`[userPageInstance1, userPageInstance2, usePageInstance3]`).

They all share the same provider (e.g. `meProvider.autoDispose`). How to scope meProvider, so each copy of it will die, when corresponding page is closed?

Example:
I pop userPageInstance3, it destroys its `meProvider`, but instance1 and instance2 still have their UNIQUE instance of provider.

With provider it's easily achieved. Just use `Provider<User>(create: ...)` on each UserPage

**Comment by TimWhiting** [comment, marked as answer]
ProviderScope(overrides: [meProvider], child: UserPage())
https://dartpad.dev/?id=b5b6826f089d80c9b1770ec4efe4267e
